Calculator/CalculatorLibrary/CalculatorLibrary.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics;
+using Newtonsoft.Json;
+
+namespace CalculatorLibrary
+{
+    public class Calculator
+    {
+        JsonWriter writer;
+
+        // Constructor incl opening Json Writer & file for log of operations
+        public Calculator()
+        {
+            StreamWriter logFile = File.CreateText("calculatorlog.json");
+            logFile.AutoFlush = true;
+            writer = new JsonTextWriter(logFile);
+            writer.Formatting = Formatting.Indented;
+            writer.WriteStartObject();
+            writer.WritePropertyName("Operations");
+            writer.WriteStartArray();
+        }
+
+        public double DoOperation(double num1, double num2, string op)
+        {
+            double result = double.NaN;  // Default value is "not-a-number" which we use if an operation such as division could result in an error.
+
+            writer.WriteStartObject();
+            writer.WritePropertyName("Operand1");
+            writer.WriteValue(num1);
+            writer.WritePropertyName("Operand2");
+            writer.WriteValue(num2);
+            writer.WritePropertyName("Operation");
+          
+
+            // Use a switch statement to do the maths
+            switch (op)
```

Comment:
I like to use 

```cs
switch(op.ToLowerInvariant())
```

when doing switches against user input because users are stupid